python/private/pypi/pypi_repo_utils.bzl

```diff
@@ -110,7 +123,7 @@ def _execute_checked(mrctx, *, srcs, **kwargs):
         # This will ensure that we will re-evaluate the bzlmod extension or
         # refetch the repository_rule when the srcs change. This should work on
         # Bazel versions without `mrctx.watch` as well.
-        repo_utils.watch(mrctx.path(src))
+        repo_utils.watch(mrctx, mrctx.path(src))
```

Comment:
Ahh, hah. I see how this got past CI. We pass along *args and do a conditional check for backwards compatiblity, and do nothing if it doesn't look supported.

It probably makes sense to check type(mrctx) in the watch function to help catch this in the future; can go into another PR, though